File: cpp/practice/armstrong.cpp
#include<iostream>

using namespace std;

int main(){

	int num,rem,result=0,temp;

	cout<<"Enter a number:";
	cin>>num;

	temp=num;

	while(temp!=0){

		rem=temp%10;

		result+=rem*rem*rem;

		temp/=10;

	}

	if(num==result)

		cout<<"This is a armstrong number";

	else

		cout<<"This isn't a armstrong number";




	return 0;
}

File: java/armstrong.java
import java.util.*;

class Main{

	public static void main(String args[]){

		Scanner sn=new Scanner(System.in);
		Main obj=new Main();
		int a;

		System.out.print("Enter a number: ");
		a=sn.nextInt();

		obj.armstrong(a);

		//System.out.println(+a);

	}

	void armstrong(int num){

		int temp,arm=0,rem;
		temp=num;
		while(temp!=0){

			rem=temp%10;
			arm=arm+rem*rem*rem;
			temp=temp/10;

		}

		if(num==arm)
			System.out.println("The number is armstrong.");

		else
			System.out.println("The number isn't armstrong.");
	}

}
